app/ide-desktop/lib/dashboard/src/hooks/searchParamsState.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * @file
+ *
+ * Search params state hook store a value in the URL search params.
+ */
+import * as React from 'react'
+
+import * as reactRouterDom from 'react-router-dom'
+
+import * as eventCallback from '#/hooks/eventCallback'
+
+import * as safeJsonParse from '#/utilities/safeJsonParse'
+
+import * as lazyMemo from './useLazyMemo'
```

Comment:
import should be absolute

Reply:
Why, though?

Reply:
1. the paths aren't that long - and even then i'd prefer *long* paths over *confusing* ones.
  - especially considering, if the paths really *are* that long, then a lot of your relative imports will be `../../../../foo`. and so it'lll be hard to figure out how far up the tree you really went.
2. not sure using relative paths helps here. also there are eslint plugins to enforce this regardless of path convention.
3. i'm not sure how using relative paths solves this. plus *barrel imports in general* cause circular deps problems, but more importantly they often make HMR invalidate way too many things. so i'm not sure whether there's a tangible benefit to using barrel exports.
4. indeed a problem with refactoring tooling. it also applies in IDEs that lack refactoring tooling completely, but yes it's true that most IDEs have good support for moving files around, so this is mostly a non-issue.

Reply:
re: very long paths - consider that what you're sacrificing for short paths is clarity:
using absolute imports, you know *exactly* how the file is categorized (in our case, things like: is it a component? is it dashboard specific?)

Reply:
Once again:

# `I'm not on a 'only relative paths` side. I'm rather choose to balance between and use the right tool for the job.`

Reply:
1. Once again: I'm not on a 'only relative paths` side. I'm rather choose to balance between and use the right tool for the job.
2. they will help since you get an error that the dependency required by the file you moved doesn't exist in place anymore.
3. `i'm not sure how using relative paths solves this.` -  `./a.js` imports `./b.js`, `i'm not sure how using relative paths solves this. plus barrel imports in general cause circular deps problems` - it's just not always true, they could lead to these problems, but it's usually happen when you poorly structure your code. `but more importantly they often make HMR invalidate way too many things.` - not a big problem with vite(you rarely edit these files). `so i'm not sure whether there's a tangible benefit to using barrel exports`  - it's apart from the current discussion

` you know exactly how the file is categorized (in our case, things like: is it a component? is it dashboard specific?)` we already know this from the context of the file.

`Once again: I'm not on a 'only relative paths` side. I'm rather choose to balance between and use the right tool for the job.`

Reply:
true, but i think it makes sense to pick one for consistency. iirc wojciech prefers absolute paths, and i don't think there's any major *downsides* of always using absolute paths